### ISSUE TITLE
Well......

### ISSUE THREAD

**Issue by perathambkk** [jihoonko/elicit#1]

**Comment by perathambkk** [comment]
From Mahanta, Mohammad Shahin, and Konstantinos N. Plataniotis. "Ranking 2DLDA features based on fisher discriminance." 2014 IEEE International Conference on Acoustics, Speech and Signal Processing (ICASSP). IEEE, 2014. who cited us.

<img width="663" height="282" alt="Image" src="https://github.com/user-attachments/assets/6658a025-6c3c-41b3-b74d-5bbac41b1de0" />

**Comment by perathambkk** [comment]
Your proposed method is also 100% exact our method....

**Comment by perathambkk** [comment]
(2D)2LDA ~ Tucker with Fisher's LDA objective, just an extension on TensorFace (Tucker).

as in Lu, Haiping, **Konstantinos N. Plataniotis**, and Anastasios N. Venetsanopoulos. "A survey of multilinear subspace learning for tensor data." Pattern Recognition 44.7 (2011): 1540-1551. clearly provides the taxonomy.

**Comment by jihoonko** [comment]
Hello,

Thank you for reaching out and bringing this paper to my attention.

First, I'd like to ask for your understanding regarding my limited availability. Since graduating from KAIST, I have been working in the industry.

I was not previously aware of the paper you mentioned, so I will need some time to review it. I will get back to you with my thoughts after I've had a chance to take a look.

Thank you for your patience.

**Comment by jihoonko** [comment]
However, when I skimmed the paper, I found that the meaning of 'Fisher' is quite different from our paper and the mentioned papers (please let me know if I am misunderstanding):

2DLDA: Uses the Fisher Score (discriminative criterion) to reorder and select eigenvectors (bases). It is a supervised feature selection method, based on maximizing the between-class / within-class scatter ratio, aiming to improve classification accuracy.
ELiCiT: **Its core method does not involve Fisher at all.** Only in the extended version (TFW-qELICIT) does it use empirical Fisher information as a loss weighting factor for neural network compression (for just reconstructing the parameters of the model). That’s not a discriminative criterion, but a weighting scheme.

**Comment by perathambkk** [comment]
It is a classification objective function. Fisher score is when it is being used in 1d as a raw measure which can also be used for feature selection as a filter. Features in the sense of the feature space. Feature selection includes weighting. The framework involves selecting features on indices are the same and the clustering step is very much what Mahanta proposed. For Tucker equivalent, the core tensor normally is being summed over rows and columns and sorted. (2D)2LDA is Tucker with a classification objective function as in the survey by Plataniotis. (2D)2LDA solves the ALS on alternating generalized eigenvalue problems so it does not return a core tensor like Tucker but the vectors of eigenvalues. I was the first in 2009 to discard the values and recreate a core tensor similar to the Tucker decomposition and also proposed a new basis selection scheme based on index selection. The implications are most core tensors or eigenvalues from ALS or tensor decomposition solutions are wrong by the ordering scheme and can be reordered. It empirically disproved the ordering axiom since I also used the same objective function for both the decomposition and reordering. On 2 Sep BE 2568, at 22:24, Jihoon Ko ***@***.***> wrote:﻿jihoonko left a comment (jihoonko/elicit#1)
However, when I skimmed the paper, I found that the meaning of 'Fisher' is quite different from our paper and the mentioned papers:
2DLDA: Uses the Fisher Score (discriminative criterion) to reorder and select eigenvectors (bases). It is a supervised feature selection method, based on maximizing the between-class / within-class scatter ratio, aiming to improve classification accuracy.
ELiCiT: Its core method does not involve Fisher at all. Only in the extended version (TFW-qELICIT) does it use empirical Fisher information as a loss weighting factor for neural network compression. That’s not a discriminative criterion, but a weighting scheme.

**Comment by perathambkk** [comment]
The basis selection in my proposed method was a greedy stepwise selection and it selects not by alternating, sometimes many bases from a specific mode are selected in consecutive which is different from ALS or any other tensor solvers which optimize one mode at a time always. On 3 Sep BE 2568, at 01:17, Peratham Wiriyathammabhum ***@***.***> wrote:﻿It is a classification objective function. Fisher score is when it is being used in 1d as a raw measure which can also be used for feature selection as a filter. Features in the sense of the feature space. Feature selection includes weighting. The framework involves selecting features on indices are the same and the clustering step is very much what Mahanta proposed. For Tucker equivalent, the core tensor normally is being summed over rows and columns and sorted. (2D)2LDA is Tucker with a classification objective function as in the survey by Plataniotis. (2D)2LDA solves the ALS on alternating generalized eigenvalue problems so it does not return a core tensor like Tucker but the vectors of eigenvalues. I was the first in 2009 to discard the values and recreate a core tensor similar to the Tucker decomposition and also proposed a new basis selection scheme based on index selection. The implications are most core tensors or eigenvalues from ALS or tensor decomposition solutions are wrong by the ordering scheme and can be reordered. It empirically disproved the ordering axiom since I also used the same objective function for both the decomposition and reordering. On 2 Sep BE 2568, at 22:24, Jihoon Ko ***@***.***> wrote:﻿jihoonko left a comment (jihoonko/elicit#1)
However, when I skimmed the paper, I found that the meaning of 'Fisher' is quite different from our paper and the mentioned papers:
2DLDA: Uses the Fisher Score (discriminative criterion) to reorder and select eigenvectors (bases). It is a supervised feature selection method, based on maximizing the between-class / within-class scatter ratio, aiming to improve classification accuracy.
ELiCiT: Its core method does not involve Fisher at all. Only in the extended version (TFW-qELICIT) does it use empirical Fisher information as a loss weighting factor for neural network compression. That’s not a discriminative criterion, but a weighting scheme.

**Comment by perathambkk** [comment]
(The basis selection is in contrast to the polyadic forms.) On 3 Sep BE 2568, at 01:29, Peratham Wiriyathammabhum ***@***.***> wrote:﻿The basis selection in my proposed method was a greedy stepwise selection and it selects not by alternating, sometimes many bases from a specific mode are selected in consecutive which is different from ALS or any other tensor solvers which optimize one mode at a time always. On 3 Sep BE 2568, at 01:17, Peratham Wiriyathammabhum ***@***.***> wrote:﻿It is a classification objective function. Fisher score is when it is being used in 1d as a raw measure which can also be used for feature selection as a filter. Features in the sense of the feature space. Feature selection includes weighting. The framework involves selecting features on indices are the same and the clustering step is very much what Mahanta proposed. For Tucker equivalent, the core tensor normally is being summed over rows and columns and sorted. (2D)2LDA is Tucker with a classification objective function as in the survey by Plataniotis. (2D)2LDA solves the ALS on alternating generalized eigenvalue problems so it does not return a core tensor like Tucker but the vectors of eigenvalues. I was the first in 2009 to discard the values and recreate a core tensor similar to the Tucker decomposition and also proposed a new basis selection scheme based on index selection. The implications are most core tensors or eigenvalues from ALS or tensor decomposition solutions are wrong by the ordering scheme and can be reordered. It empirically disproved the ordering axiom since I also used the same objective function for both the decomposition and reordering. On 2 Sep BE 2568, at 22:24, Jihoon Ko ***@***.***> wrote:﻿jihoonko left a comment (jihoonko/elicit#1)
However, when I skimmed the paper, I found that the meaning of 'Fisher' is quite different from our paper and the mentioned papers:
2DLDA: Uses the Fisher Score (discriminative criterion) to reorder and select eigenvectors (bases). It is a supervised feature selection method, based on maximizing the between-class / within-class scatter ratio, aiming to improve classification accuracy.
ELiCiT: Its core method does not involve Fisher at all. Only in the extended version (TFW-qELICIT) does it use empirical Fisher information as a loss weighting factor for neural network compression. That’s not a discriminative criterion, but a weighting scheme.

**Comment by jihoonko** [comment]
Regarding “selecting features on indices,” attaching per-mode latent features to indices is standard in tensor models (CP decomposition is one of the very typical examples). What distinguishes ELiCiT is structural: it is not an alternating polyadic solver; it is order-free among features (of course, since it is order-free, we do not perform reordering at all, and we do not perform stepwise selection), and uses "END-TO-END" (i.e., there is no separate training stage for quantization) clustering-based quantization to optimize the compression–fitness trade-off rather than discriminative basis selection.
In our paper, the clustering step is the quantization mechanism for compression—not a procedure to mine discriminative features. We do not perform any explicit discriminative feature extraction, as we do not consider clustering-related term in our loss(objective) function. Furthermore, our method follows standard end-to-end clustering-based quantization in lots of deep learning researches, including the papers listed below.
Our primary objective is reconstruction fitness and model size, not a discriminative criterion; we do not explicitly optimize for “extracting the most discriminant features” during clustering-based quantization.

+ Note that ELiCiT without quantization still outperformed decomposition-based baseline (nearly 2X compact compared to the competitors with similar fitness score)

--

T. Chen, L. Li, and Y. Sun, "Differentiable product quantization for end-to-end embedding compression," in ICML, 2020.
A. Oord, O. Vinyals, and K. Kavukcuoglu, "Neural Discrete Representation Learning", in NeurIPS, 2017.

**Comment by perathambkk** [comment]
Well . I don’t think you provide any sound explanations at all now. How can a Tensor Decomposition not for compression and unrelated? Also, you contradicted yourself on claims from your own papers. Projection on smaller numbers of bases provides compression and reordering is mentioned in your papers with exact wordings. The violations on ordered bases are the same as new measures or scoring. I guess this is clear. Please properly cite the preceding works or just retract your works. End-to-end means something else for deep learning literatures. Shame on you! On 7 Sep BE 2568, at 17:25, Jihoon Ko ***@***.***> wrote:﻿jihoonko left a comment (jihoonko/elicit#1)
Regarding “selecting features on indices,” attaching per-mode latent features to indices is standard in tensor models (CP decomposition is one of the very typical examples). What distinguishes ELiCiT is structural: it is not an alternating polyadic solver; it is order-free among features (of course, since it is order-free, we do not perform reordering at all, and we do not perform stepwise selection), and uses "END-TO-END" (i.e., there is no separate training stage for quantization) clustering-based quantization to optimize the compression–fitness trade-off rather than discriminative basis selection.
In our paper, the clustering step is the quantization mechanism for compression—not a procedure to mine discriminative features. We do not perform any explicit discriminative feature extraction, as we do not consider this point in our objective function. Furthermore, our method follows standard end-to-end clustering-based quantization in lots of deep learning researches, including the papers listed below.
Our primary objective is reconstruction fitness and model size, not a discriminative criterion; we do not explicitly optimize for “extracting the most discriminant features” during clustering-based quantization.
--
T. Chen, L. Li, and Y. Sun, "Differentiable product quantization for end-to-end embedding compression," in ICML, 2020.
A. Oord, O. Vinyals, and K. Kavukcuoglu, "Neural Discrete Representation Learning", in NeurIPS, 2017.

**Comment by perathambkk** [comment]
The selection operator can be viewed as a binary constant on the set of bases. Most standard quantization can be done with many algorithms, a stepwise selection is just an example of them. On 7 Sep BE 2568, at 20:40, Peratham Wiriyathammabhum ***@***.***> wrote:﻿Well . I don’t think you provide any sound explanations at all now. How can a Tensor Decomposition not for compression and unrelated? Also, you contradicted yourself on claims from your own papers. Projection on smaller numbers of bases provides compression and reordering is mentioned in your papers with exact wordings. The violations on ordered bases are the same as new measures or scoring. I guess this is clear. Please properly cite the preceding works or just retract your works. End-to-end means something else for deep learning literatures. Shame on you! On 7 Sep BE 2568, at 17:25, Jihoon Ko ***@***.***> wrote:﻿jihoonko left a comment (jihoonko/elicit#1)
Regarding “selecting features on indices,” attaching per-mode latent features to indices is standard in tensor models (CP decomposition is one of the very typical examples). What distinguishes ELiCiT is structural: it is not an alternating polyadic solver; it is order-free among features (of course, since it is order-free, we do not perform reordering at all, and we do not perform stepwise selection), and uses "END-TO-END" (i.e., there is no separate training stage for quantization) clustering-based quantization to optimize the compression–fitness trade-off rather than discriminative basis selection.
In our paper, the clustering step is the quantization mechanism for compression—not a procedure to mine discriminative features. We do not perform any explicit discriminative feature extraction, as we do not consider this point in our objective function. Furthermore, our method follows standard end-to-end clustering-based quantization in lots of deep learning researches, including the papers listed below.
Our primary objective is reconstruction fitness and model size, not a discriminative criterion; we do not explicitly optimize for “extracting the most discriminant features” during clustering-based quantization.
--
T. Chen, L. Li, and Y. Sun, "Differentiable product quantization for end-to-end embedding compression," in ICML, 2020.
A. Oord, O. Vinyals, and K. Kavukcuoglu, "Neural Discrete Representation Learning", in NeurIPS, 2017.

**Comment by perathambkk** [comment]
Plagiarism and academic dishonesty are as bad as medical malpracticing or illegal modifications in buildings. Many people do not care as long as no legal enforcement can touch them 🤷‍♂️On 7 Sep BE 2568, at 20:45, Peratham Wiriyathammabhum ***@***.***> wrote:﻿The selection operator can be viewed as a binary constant on the set of bases. Most standard quantization can be done with many algorithms, a stepwise selection is just an example of them. On 7 Sep BE 2568, at 20:40, Peratham Wiriyathammabhum ***@***.***> wrote:﻿Well . I don’t think you provide any sound explanations at all now. How can a Tensor Decomposition not for compression and unrelated? Also, you contradicted yourself on claims from your own papers. Projection on smaller numbers of bases provides compression and reordering is mentioned in your papers with exact wordings. The violations on ordered bases are the same as new measures or scoring. I guess this is clear. Please properly cite the preceding works or just retract your works. End-to-end means something else for deep learning literatures. Shame on you! On 7 Sep BE 2568, at 17:25, Jihoon Ko ***@***.***> wrote:﻿jihoonko left a comment (jihoonko/elicit#1)
Regarding “selecting features on indices,” attaching per-mode latent features to indices is standard in tensor models (CP decomposition is one of the very typical examples). What distinguishes ELiCiT is structural: it is not an alternating polyadic solver; it is order-free among features (of course, since it is order-free, we do not perform reordering at all, and we do not perform stepwise selection), and uses "END-TO-END" (i.e., there is no separate training stage for quantization) clustering-based quantization to optimize the compression–fitness trade-off rather than discriminative basis selection.
In our paper, the clustering step is the quantization mechanism for compression—not a procedure to mine discriminative features. We do not perform any explicit discriminative feature extraction, as we do not consider this point in our objective function. Furthermore, our method follows standard end-to-end clustering-based quantization in lots of deep learning researches, including the papers listed below.
Our primary objective is reconstruction fitness and model size, not a discriminative criterion; we do not explicitly optimize for “extracting the most discriminant features” during clustering-based quantization.
--
T. Chen, L. Li, and Y. Sun, "Differentiable product quantization for end-to-end embedding compression," in ICML, 2020.
A. Oord, O. Vinyals, and K. Kavukcuoglu, "Neural Discrete Representation Learning", in NeurIPS, 2017.
